src/app/order/OrderStatus.tsx

```diff
@@ -11,29 +11,18 @@ export interface OrderStatusProps {
     order: Order;
 }
 
+type PaymentWithMandate = GatewayOrderPayment & Required<Pick<GatewayOrderPayment, 'mandate' | 'methodId'>>;
+
+const isPaymentWithMandate = (payment: GatewayOrderPayment | GiftCertificateOrderPayment): payment is PaymentWithMandate =>
+    !!payment.methodId && ('mandate' in payment && !!payment.mandate);
+
 const OrderStatus: FunctionComponent<OrderStatusProps> = ({
     order,
     supportEmail,
     supportPhoneNumber,
 }) => {
 
-    const getMandateProvider = useCallback(() => {
-        return order?.payments?.[0].description;
-    }, [order]);
-
-    const getMandateTextId = useCallback(() => {
-        const Mandates = [
-            { method: 'Stripe (SEPA)', value: 'sepa_link_text' },
-            { method: 'OXXO (via Checkout.com)', value: 'oxxo_link_text' },
-            { method: 'OXXO (via Stripe)', value: 'oxxo_link_text'},
-            { method: 'Boleto Bancário (via Checkout.com)', value: 'boleto_link_text' },
-            { method: 'Boleto (via Stripe)', value: 'boleto_link_text'},
-        ];
-
-        const mandateText = Mandates.find(pair => pair.method === order?.payments?.[0].description);
-
-        return mandateText ? mandateText.value : 'mandate_link_text';
-    }, [order]);
+    const paymentsWithMandates = order?.payments?.filter(isPaymentWithMandate) || [];
```

Comment:
`order` seems to be mandatory, doesn't need the optional check

Reply:
done :+1: